src/measure_theory/measure/haar_of_basis.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import measure_theory.measure.haar
+import analysis.inner_product_space.pi_L2
+
+/-!
+# Additive Haar measure constructed from a basis
+
+Given a basis of a finite-dimensional real vector space, we define the corresponding Lebesgue
+measure, which gives measure `1` to the parallelogram spanned by the basis.
+
+## Main definitions
+
+* `parallelogram v` is the parallelogram spanned by a finite family of vectors.
+* `basis.parallelogram` is the parallelogram associated to a basis, seen as a compact set with
+nonempty interior.
+* `basis.add_haar` is the Lebesgue measure associated to a basis, giving measure `1` to the
+corresponding parallelogram.
+
+In particular, we declare a `measure_space` instance on any finite-dimensional inner product space,
+by using the Lebesgue measure associated to some orthonormal basis (which is in fact independent
+of the basis).
+-/
+
+open set topological_space measure_theory measure_theory.measure finite_dimensional
+open_locale big_operators
+
+noncomputable theory
+
+variables {ι ι' E F : Type*} [fintype ι] [fintype ι']
+
+section add_comm_group
+
+variables [add_comm_group E] [module ℝ E] [add_comm_group F] [module ℝ F]
+
+/-- The parallelogram spanned by a finite family of vectors. -/
+def parallelogram (v : ι → E) : set E :=
+(λ (t : ι → ℝ), ∑ i, t i • v i) '' (Icc 0 1)
+
+lemma image_parallelogram (f : E →ₗ[ℝ] F) (v : ι → E) :
+  f '' (parallelogram v) = parallelogram (f ∘ v) :=
+begin
+  simp only [parallelogram, ← image_comp],
+  congr' 1,
+  ext t,
+  simp only [function.comp_app, linear_map.map_sum, linear_map.map_smulₛₗ, ring_hom.id_apply],
+end
+
+/-- Reindexing a family of vectors does not change their parallelogram. -/
+@[simp] lemma parallelogram_comp (v : ι → E) (e : ι' ≃ ι) :
```

Comment:
```suggestion
@[simp] lemma parallelogram_comp_equiv (v : ι → E) (e : ι' ≃ ι) :
```

Reply:
It's funny, that's the name I had in a previous version, and then I removed the `equiv` because there was no ambiguity. I will add it back.